### PR TITLE
feat: delete invalid case

### PR DESCRIPTION
**Pull request by YongwuHe**

### CODE REVIEW

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java [hunk 1]
```diff
@@ -4,20 +4,35 @@
...
                 return;
             }
+            context.setInvalidCase(true);
+            Map<String, String> invalidCase = new ConcurrentHashMap<>(3);
+            invalidCase.put(APP_ID, System.getProperty(AREX_SERVICE_NAME));
+            invalidCase.put(RECORD_ID, recordId);
+            invalidCase.put(REASON, invalidReason.name());
+            DataService.INSTANCE.invalidCase(Serializer.serialize(invalidCase));
             LogManager.warn("invalidCase",
                     StringUtil.format("invalid case: recordId: %s operation: %s", recordId,  operationName));
```
Comment (lucas-myx) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java:35]: suggest add invalidReason to log

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/InvalidReason.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package io.arex.inst.runtime.util;
+
+public enum InvalidReason {
+    FAST_REJECT,
```
Comment (lucas-myx) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/InvalidReason.java:4]: `FAST_REJECT` is a common state，which will be switched to first after queue overflow and service exceptions, so here should be `SERVICE_EXCEPTION` ?

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java [hunk 3]
```diff
@@ -4,20 +4,35 @@
...
             final ArexContext context = ContextManager.getRecordContext(recordId);
             if (context == null || context.isInvalidCase()) {
                 return;
             }
+            context.setInvalidCase(true);
+            Map<String, String> invalidCase = new ConcurrentHashMap<>(3);
+            invalidCase.put(APP_ID, System.getProperty(AREX_SERVICE_NAME));
+            invalidCase.put(RECORD_ID, recordId);
+            invalidCase.put(REASON, invalidReason.name());
+            DataService.INSTANCE.invalidCase(Serializer.serialize(invalidCase));
```
Comment (mr3) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java:33]: Suggest directly use string format `{"appId":"%s","recordId":"%s","reason":"%s"}`, no need use serialize

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/InvalidReason.java (hunk 2 above)
Comment (mr3) [reply arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/InvalidReason.java:4]: Duplicate with class `io.arex.foundation.model.DecelerateReasonEnum`

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java [hunk 4]
```diff
@@ -4,20 +4,35 @@
...
+    private static final String RECORD_ID = "recordId";
+    private static final String APP_ID = "appId";
+    private static final String AREX_SERVICE_NAME = "arex.service.name";
+    private static final String REASON = "reason";
+
     private CaseManager() {
     }
 
-    public static void invalid(String recordId, String operationName) {
+    public static void invalid(String recordId, String operationName, InvalidReason invalidReason) {
```
Comment (mr3) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/CaseManager.java:22]: Suggest change to `String invalidReason`

**Review by mr3 — APPROVED** [review]

**Review by mr3 — APPROVED** [review]

**Review by lucas-myx — APPROVED** [review]

**Pull request merged**